packages/clang5.rb

```diff
@@ -0,0 +1,46 @@
+require 'package'
+
+class Clang5 < Package
+  description 'C language family frontend for LLVM.'
+  homepage 'http://clang.llvm.org/'
+  version '5.0.1'
+  source_url 'http://releases.llvm.org/5.0.1/cfe-5.0.1.src.tar.xz'
+  source_sha256 '135f6c9b0cd2da1aff2250e065946258eb699777888df39ca5a5b4fe5e23d0ff'
+
+  depends_on 'buildessential'
+  depends_on 'cmake'
+  depends_on 'llvm5'
+  depends_on 'clang5_libcpp'
+  depends_on 'clang5_openmp'
+  depends_on 'llvm5_compiler_rt'
+  depends_on 'llvm5_polly'
+  depends_on 'llvm5_unwind'
+  depends_on 'lld5'
+
+  def self.build
+    system "mkdir -p mybuilddir"
+    Dir.chdir "mybuilddir" do
+ 	    system "cmake -DBUILD_SHARED_LIBS=ON -DCMAKE_BUILD_TYPE=Release .."
+        system "cmake --build . --  -j#{CREW_NPROC}"
+    end
+  end
+
+  def self.install
+    Dir.chdir "mybuilddir" do
+      system "cmake -DCMAKE_INSTALL_PREFIX=#{CREW_DEST_DIR}/usr/local -P cmake_install.cmake"
+      FileUtils.cd("#{CREW_DEST_PREFIX}/bin") do
+        system "echo '#!/bin/bash' > clang++w"
+        system "echo 'GCCVERSION=`gcc --version | grep ^gcc | sed '\\''s/^.* //g'\\''`' >> clang++w"
+        system "echo 'TARGET=`gcc -dumpmachine`' >> clang++w"
+        system "echo '/usr/local/bin/clang++ -I/usr/local/include/c++/v1 -B/usr/local/lib/gcc/${TARGET}/${GCCVERSION} -lc++ $@' >> clang++w"
+        system "chmod 755 clang++w"
+        system "echo '#!/bin/bash' > clangw"
+        system "echo 'GCCVERSION=`gcc --version | grep ^gcc | sed '\\''s/^.* //g'\\''`' >> clangw"
+        system "echo 'TARGET=`gcc -dumpmachine`' >> clangw"
+        system "echo '/usr/local/bin/clang -B/usr/local/lib/gcc/${TARGET}/${GCCVERSION} $@' >> clangw"
```

Comment:
Again, you will want to escape both `$` and `@`.

Reply:
Actually, you probably will want to quote the `$@` also.  Sometimes arguments have spaces.  So the final string should look like: `\"\$\@\"`.  So much escaping, I know.

Reply:
See above. The file seems to end up being created properly in my tests.